src/v/kafka/server/group.h

```diff
@@ -139,6 +140,20 @@ class group {
         // https://github.com/vectorizedio/redpanda/issues/1181
     };
 
+    struct offset_metadata_with_probe {
+        offset_metadata metadata;
+        group_offset_probe probe;
+
+        offset_metadata_with_probe(
+          offset_metadata _metadata,
+          const kafka::group_id& group_id,
+          const model::topic_partition& tp)
+          : metadata(std::move(_metadata))
+          , probe(metadata.offset) {
```

Comment:
what guarantees that this reference to the offset will remain valid? for example if I made a copy of this structure and deleted the original version then the copy would contain a reference to freed memory.

Reply:
We can't create copy of `offset_metadata_with_probe` because we can't copy probe
That is why probe can't containt reference to metadata from another `offset_metadata` structure and can only has reference to  `offset_metadata` in its `offset_metadata_with_probe` structure

Reply:
what if you `std::move` an instance of `offset_metadata_with_probe`? the moved-to instance will contain a reference to the wrong `offset`, right?